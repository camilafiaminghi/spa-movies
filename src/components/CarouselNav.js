import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import './CarouselNav.scss';

const CarouselNav = ({handleNavigation, left, right}) => {

	return (
		<Fragment>
			<button
				onClick={() => handleNavigation('left')}
				disabled={!left}
				className={left ? 'carousel-button carousel-left active' : 'carousel-button carousel-left'}
				tabIndex="0"
				type="button">nav to left</button>
			<button
				onClick={() => handleNavigation('right')}
				disabled={!right}
				className={right ? 'carousel-button carousel-right active' : 'carousel-button carousel-right'}
				tabIndex="0"
				type="button">nav to right</button>
		</Fragment>
	)
}

CarouselNav.propTypes = {
	handleNavigation: PropTypes.func.isRequired,
	left: PropTypes.bool.isRequired,
	right: PropTypes.bool.isRequired
}

export default CarouselNav
