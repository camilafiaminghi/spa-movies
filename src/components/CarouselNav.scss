@import "./../scss/variables.scss";
@import "./../scss/mixins.scss";

.carousel-button {
	overflow: hidden;
	position: absolute;
	top: 0;
	bottom: 0;
	color: $color-default;
	opacity: 0.5;
	transition: all 0.3s linear 0s;
	display: none;
	outline: none;

	&:after {
		display: block;
		font-size: 75px;
		padding: 0 0.2em;
	}

	&:hover {
		opacity: 1;
	}

	&.active {
		@include breakpoint($sm-device-min) {
			display: block;
		}
	}

	&.carousel-left {
		left: 0;

		&:after {
			content: "⟨"
		}
	}

	&.carousel-right {
		right: 0;

		&:after {
			content: "⟩"
		}
	}
}
