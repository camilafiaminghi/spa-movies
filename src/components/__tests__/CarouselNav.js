import React from 'react'
import {
	render,
	fireEvent,
	cleanup
} from '@testing-library/react'
import CarouselNav from './../CarouselNav'

const handleNavigation = jest.fn()
const right = true, left = true

describe('<CarouselNav />', () => {

	afterEach(cleanup)

	it('renders the component', () => {
		const container = render(<CarouselNav handleNavigation={handleNavigation} left={left} right={right} />)
		expect(container.firstChild).toMatchSnapshot()
	})

	it('left button disabled', () => {
		const { getByText } = render(<CarouselNav handleNavigation={handleNavigation} left={false} right={right} />)
		expect(getByText('nav to left')).toHaveAttribute('disabled')
	})

	it('right button disabled', () => {
		const { getByText } = render(<CarouselNav handleNavigation={handleNavigation} left={left} right={false} />)
		expect(getByText('nav to right')).toHaveAttribute('disabled')
	})

	it('fire event click', () => {
		const { getByText } = render(<CarouselNav handleNavigation={handleNavigation} left={left} right={right} />)

		fireEvent.click(getByText('nav to left'))
		expect(handleNavigation).toHaveBeenCalledWith('left')

		fireEvent.click(getByText('nav to right'))
		expect(handleNavigation).toHaveBeenCalledWith('right')
	})
})
