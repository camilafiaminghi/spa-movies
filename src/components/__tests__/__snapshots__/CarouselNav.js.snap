// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselNav /> renders the component 1`] = `undefined`;
